azure-quantum/azure/quantum/projectq/backends/ionq.py

```diff
@@ -108,20 +131,29 @@ def run(self, name=None, **kwargs):
 
     def _run(self):
         """
-        Overriding Projectq run method with empty function.
-        This disables circuit execution using default ProjectQ logic. Use engine.run() to submit job to Azure Quantum.
+        Run a ProjectQ circuit and wait until it is done.
         """
-        pass
+        job = self.submit_job()
```

Comment:
Supporting `eng.flush()` method might cause other issues.

Before ProjectQ MainEngine destroyed, it flushes the existing circuit to IONQ APIs. So, if we add support for flush method, user might submit job without calling flush method.
Also, when users submit job using `eng.submit_job()` method, client will submit job to Azure Quantum twice (first time because of user action and second time when MainEngine destroyed).

Let me know your thoughts on this.